### PR TITLE
Leaves - Georgina

### PR DESCRIPTION
**Pull request by geomsb**
# Calculator
Congratulations! You're submitting your assignment.

## Comprehension Questions

| Question | Answer |
| --- | --- |
| What went well in your code style, such as indentation, spacing, variable names, readability, etc.? What was lacking? | The indentation and the variable names were good.
| How did your code keep track of user input?	|  I created variables.
| How did your code determine what operation to perform? | I used a conditional statement to evaluate the input from the user.
| What opportunities exist to create small methods for this project? | I created a method for each operation.
| In the next project, what would you change about your process? What would you keep doing? | I would keep breaking the problem in smaller problems. I would take more time designing my code before I start writing.

### CODE REVIEW

File: calculator.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+def modulo(num_one, num_two)
+  return num_one % num_two
+end
+
+# The program handles divide when attempting to divide by zero
+def divide(num_one, num_two)
+  if num_two == 0
+    until num_two != 0
+      puts "You can't divide by zero. Enter the second number again: "
+      num_two = gets.chomp.to_i
```
Comment (dHelmgren) [inline calculator.rb]: Looks like a property of ruby has bit you here. You can't override the value of a parameter in ruby. See the following output from your program:

```
Please choose a valid operator (name or symbol): 
/
Enter the first number: 
fjkajflkajkdf
Enter a valid first number: 
9
Enter the second number 
dafjk
Enter a valid second number: 
0
You can't divide by zero. Enter the second number again: 
8
9 / 0 = 1
```

File: calculator.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    end
+  end
+    return num_one / num_two
+end
+
+if operator == "add" || operator == "+" 
+  result = add(num_one_string.to_i, num_two_string.to_i)
+  puts "#{num_one_string}" " + " "#{num_two_string}" " = " "#{result}"
+elsif operator == "substract" || operator == "-"
```
Comment (dHelmgren) [inline calculator.rb:68]: This works, though I am surprised you didn't go for a case statement!

**Comment by dHelmgren** [comment]
# Calculator
## What We're Looking For



Feature  | Feedback
---       | ---
Readable code with consistent indentation | yes
Practices using variables appropriately | yes
Practices using conditionals appropriately | yes
If any, practices iteration appropriately | yes
If any, practices using custom methods appropriately | yes
Takes in two numbers and an operator and can perform addition | yes
Takes in two numbers and an operator and can perform subtraction | yes
The program handles divide when attempting to divide by zero | no, see comment!

--- | ---
Overall | Solid work! With the exception of a problem with divide, this demonstrates that you met the learning goals for this assignment. Remember to test any code that you write carefully, it's easy to miss when an edge case goes wrong.

File: calculator.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  result = substract(num_one_string.to_i, num_two_string.to_i)
+  puts "#{num_one_string}" " - " "#{num_two_string}" " = " "#{result}"
+elsif operator == "multiply" || operator == "*"
+  result = multiply(num_one_string.to_i, num_two_string.to_i)
+  puts "#{num_one_string}" " * " "#{num_two_string}" " = " "#{result}"
+elsif operator == "divide" || operator == "/"
+  result = divide(num_one_string.to_i, num_two_string.to_i)
+  puts "#{num_one_string.to_i}" " / " "#{num_two_string.to_i}" " = " "#{result}"
```
Comment (dHelmgren) [inline calculator.rb:76]: You get a new input inside your divide method, but the num_two out here is still the old value, leading to input like this:

```
Please choose a valid operator (name or symbol): 
/
Enter the first number: 
fjkajflkajkdf
Enter a valid first number: 
9
Enter the second number 
dafjk
Enter a valid second number: 
0
You can't divide by zero. Enter the second number again: 
8
9 / 0 = 1
```
